lms/djangoapps/courseware/views/index.py

```diff
@@ -147,7 +147,7 @@ def get(self, request, course_id, chapter=None, section=None, position=None):
                         # If the user is considered enrolled show the default XBlock student_view.
                         pass
 
-                self.is_staff = has_access(request.user, 'staff', self.course)
+                self.can_masquerade = request.user.has_perm('courseware.masquerade_as_student', self.course)
```

Comment:
do we want to name things like this around the action (like can_masquerade) or mention the perm, like has_masquerade_as_student_permission?

also, if we did the former, it seems like this should be can_masquerade_as_student rather than can_masquerade? currently, if you can masquerade as a student you can also do a generic masquerade but in theory that doesn't have to be true

Reply:
I think in this case, the only type of masquerading that can happen is maquerading as students of one type or another, which is why I didn't use exactly the same terminology.

Reply:
But yes, let's make it more clear.

Reply:
Actually... I think I'm ok w/ how it is. The preview menu is how all masquerading is done. Currently, only student masquerading happens, but in theory, you could have a separate type of masquerading that would allow you to show the masquerade bar...

Reply:
ah hmm, i think in my mind masquerade_as_student meant masquerade_as_specific_student, but i guess if masquerade_as_student is the most accurate name that also encompasses masquerading as user groups then that's fine